### PR TITLE
Adds TextTag, PrimaryTextTag and SecondaryTextTag widgets

### PR DESCRIPTION
**Pull request by matuella**

### CODE REVIEW

File: lib/application/widgets/text_tag.dart [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+/// Creates a new [Text] that resembles a tag
+///
+/// It's important to no confuse this tag-styled component with the material's `Chip`, which is used in more complex
+/// - usually interactable - scenarios.
+class TextTag extends StatelessWidget {
+  const TextTag(
+    this.text, {
+    required this.backgroundColor,
+    required this.padding,
```
Comment (ggirotto) [inline lib/application/widgets/text_tag.dart:12]: I couldn't understand the use-case that needs this `padding` to be parametrized in the Widget

File: lib/application/widgets/theme/themed_text_tag.dart [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import 'package:flutter/material.dart';
```
Comment (ggirotto) [inline lib/application/widgets/theme/themed_text_tag.dart:1]: In this file as an example, both implementations use the same padding: `context.allInsets(Spacing.xxSmall)`

**Review by ggirotto — CHANGES_REQUESTED** [review]

File: lib/application/widgets/theme/themed_text_tag.dart (hunk 2 above)
Comment (matuella) [reply lib/application/widgets/theme/themed_text_tag.dart:1]: Well, it splits the responsibility of the implementation versus what the component needs.

My perspective is that if you can split things into smaller workable and independent stuff, without adding too much overhead to the code, you should do that. In this case, for instance, I don't need to force it down through the `TextTag`, as the implementers that are already exposed to a lot of distinct frameworks (like flutter hooks, riverpods, layoutr), can remain being "dirty", while the core can remain mostly pristine, only accessing fundamental frameworks, like material.

That's one of the reasons why I have split the `widgets/` folder. It's described a little bit more in ARCHITECTURE.md

Comment (ggirotto) [reply lib/application/widgets/theme/themed_text_tag.dart:1]: Got it

**Review by ggirotto — APPROVED** [review]

**Pull request merged**